# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JitActivation.h
# Commit: 270961083337
# Full Hash: 270961083337349dc71f3a2359b7cdc4b1c83a0f
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-20 14:55:47
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
#   All these places require short fixes to use our new scheme for getting
# ==============================================================================

diff -r db39bf2058e1 -r 270961083337 js/src/vm/JitActivation.h
--- a/js/src/vm/JitActivation.h	Tue Oct 20 10:24:40 2020 +0000
+++ b/js/src/vm/JitActivation.h	Tue Oct 20 10:24:47 2020 +0000
@@ -214,6 +214,9 @@
     return reinterpret_cast<wasm::Frame*>(
         wasm::Frame::toJitEntryCaller(packedExitFP_));
   }
+  wasm::TlsData* wasmExitTls() const {
+    return wasm::GetNearestEffectiveTls(wasmExitFP());
+  }
   void setWasmExitFP(const wasm::Frame* fp) {
     if (fp) {
       MOZ_ASSERT(!wasm::Frame::isExitOrJitEntryFP(fp));